extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -918,8 +918,9 @@ class LifecycleTask extends BackbeatTask {
                     method: 'LifecycleTask._applyTransitionRule',
                     error: err,
                 });
+            } else {
+                log.debug('transition rule applied');
```

Comment:
Not directly in the scope of this PR, though logs without any context tend to be useful only for the developer that actually put those logs in the first place while developing the code. It could be made more useful by adding at least the object info in the log.

Reply:
Good point. I've added the method, bucket name, object key, and site name in this debug log.